src/app/store/absence.reducer.ts

```diff
@@ -0,0 +1,79 @@
+import {createReducer, on} from "@ngrx/store";
+
+import {addAbsence, deleteAbsence, updateAbsence} from "./absence.action";
+import {AbsenceState} from "./absence.state";
+
+export const initialState: AbsenceState[] = [];
+
+const findOrCreateYearState = (state: AbsenceState[], year: number): AbsenceState => {
+  const yearState = state.find(absence => absence.year === year);
+  return yearState
+    ? yearState
+    : {year, absence: [], sickCount: 0, vacationCount: 0};
+};
+
+const updateCounts = (state: AbsenceState, days: number, absenceType: string, operation: 'add' | 'subtract') => {
```

Comment:
updateCounts -> getUpdatedCounts?

the name is a bit misleading - we don't update smth in this function, we just get a data